File: AtCoder Problems/AtCoder Beginners Selection/ABC086C - Traveling/ABC086C - Traveling.go
package main

import(
	"fmt"
)

func abs(a int) int {
	if a < 0 {
		return -a
	}
	return a
}

func main() {
	var N int
	fmt.Scan(&N)

	var tPast, xPast, yPast int

	for i := 0; i < N; i++ {
		var t, x, y int
		fmt.Scan(&t, &x, &y)

		length := abs(x - xPast) + abs(y - yPast)
		time := t - tPast

		tPast, xPast, yPast = t, x, y

		if length <= time && length % 2 == time % 2{
			// 
		} else{
			fmt.Println("No")
			return
		}
	}
	fmt.Println("Yes")
}

File: AtCoder Problems/AtCoder Beginners Selection/ABC086C - Traveling/ABC086C - Traveling.py
N = int(input())

T_PAST, X_PAST, Y_PAST = 0, 0, 0

for i in range(N):
  T, X, Y = map(int, input().split())

  LENGTH = abs(X - X_PAST) + abs(Y - Y_PAST)
  TIME = T - T_PAST

  if LENGTH <= TIME and (LENGTH % 2) == (TIME % 2):
    pass
  else:
    print("No")
    exit()
  
  T_PAST, X_PAST, Y_PAST = T, X, Y

print("Yes")